frontend/src/components/phones/onboarding/RelayNumberPicker.tsx

```diff
@@ -94,10 +95,12 @@ const RelayNumberSelection = (props: RelayNumberSelectionProps) => {
   const [searchValue, setSearchValue] = useState("");
   const [relayNumberIndex, setRelayNumberIndex] = useState(0);
   const [isSearching, setIsSearching] = useState(false);
+  const [confirmationState, setConfirmationState] = useState<boolean>(false);
```

Comment:
Currently, the modal can't be closed by pressing Esc or clicking outside of the modal. To fix this, instead of a simple boolean, you'll want to use an overlayTriggerState, and pass `isOpen` to the modal. I think you used the `<AddressPickerModal>` as inspiration, so to see how it's done there, see https://github.com/mozilla/fx-private-relay/blob/fb8df4e1b8a76827b84a4afa4c3b422f685dee07/frontend/src/components/dashboard/aliases/AliasGenerationButton.tsx#L129.

Reply:
Actually, I had to implement that myself to figure out that that was the issue, so I might as well just push that code :P See this PR: https://github.com/mozilla/fx-private-relay/pull/2455